wagtail/admin/templates/wagtailadmin/shared/header.html

```diff
@@ -14,15 +14,19 @@
     - `action_icon` - icon for the 'action' button, default is 'icon-plus'
 
 {% endcomment %}
-<header class="header {% if merged %}merged{% endif %} {% if search_form %}hasform{% endif %}">
+<header class="header{% if merged %} merged{% endif %}{% if search_form %}hasform{% endif %} {{ classname }}">
```

Comment:
Also be careful here, I think you accidentally broke the existing classes
`hasform` now shows bang up against `header` if used.

```suggestion
<header class="header {{ classname }} {{ merged|yesno:" merged ," }}{{ search_form|yesno:" hasform ," }}">
```

There might be a nicer way but you could leverage the `yesno` filter to wrap the class with spaces if it it is needed https://docs.djangoproject.com/en/4.0/ref/templates/builtins/#yesno


But longer term - we may need to find a better way for this.